warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/NonShardedSplitsFile.java

```diff
@@ -153,49 +165,73 @@ private static boolean matchesFileName(String cutFileName, Path cacheFile) {
-                        if (parts.length > 1)
-                            cutPoints.add(new Text(Base64.decodeBase64(parts[1].getBytes())));
-                    } else {
-                        Collections.sort(cutPoints);
-                        splits.put(previousTableName, cutPoints.toArray(new Text[cutPoints.size()]));
-                        log.info("Adding cut points for table: " + previousTableName);
+                        isFirstLine = false;
+                    }
+                    if (!isSameTable) {
+                        saveLastSplitForTable(cutPoints, cutPointsAndLocations, splits, splitsAndLocations, previousTableName, hasSplits, hasLocations);
                         previousTableName = parts[0];
                         cutPoints.clear();
-                        if (parts.length > 1)
-                            cutPoints.add(new Text(Base64.decodeBase64(parts[1].getBytes())));
+                        cutPointsAndLocations.clear();
+                    }
+                    if (hasLocations) {
+                        cutPointsAndLocations.put(new Text(Base64.decodeBase64(parts[1].getBytes())), parts[2]);
+                    } else if (hasSplits) {
+                        cutPoints.add(new Text(Base64.decodeBase64(parts[1].getBytes())));
                     }
                 }
             } finally {
-                if (null != previousTableName) {
-                    // Add the last batch.
-                    Collections.sort(cutPoints);
-                    splits.put(previousTableName, cutPoints.toArray(new Text[cutPoints.size()]));
-                    log.info("Adding cut points for table: " + previousTableName);
-                }
+                // Add the last batch
+                saveLastSplitForTable(cutPoints, cutPointsAndLocations, splits, splitsAndLocations, previousTableName, hasSplits, hasLocations);
             }
         }
         
         public Map<String,Text[]> getSplitsByTable() {
             return splits;
         }
+        
+        public Map<String,Map<Text,String>> getSplitsAndLocationsByTable() {
+            return splitsAndLocations;
+        }
+        
+        private void saveLastSplitForTable(ArrayList<Text> cutPoints, SortedMap<Text,String> cutPointsAndLocations, Map<String,Text[]> splits,
+                        Map<String,Map<Text,String>> splitsAndLocations, String previousTableName, boolean hasSplits, boolean hasLocations) {
+            Collections.sort(cutPoints);
+            if (hasLocations) {
+                splitsAndLocations.put(previousTableName, new TreeMap(cutPointsAndLocations));
+                log.info("Adding cut points and locations for the table: " + previousTableName);
+                splits.put(previousTableName, cutPointsAndLocations.keySet().toArray(new Text[cutPointsAndLocations.size()]));
+            } else if (hasSplits) {
+                splits.put(previousTableName, cutPoints.toArray(new Text[cutPoints.size()]));
```

Comment:
As it turns out it is cheeper to do new Text[0] here instead of new Text[cutPoints.size()].  https://shipilev.net/blog/2016/arrays-wisdom-ancients/